erica/erica_legacy/elster_xml/common/electronic_steuernummer.py

```diff
@@ -0,0 +1,33 @@
+from erica.erica_legacy.elster_xml.est_mapping import generate_electronic_steuernummer
+from erica.erica_legacy.pyeric.eric import get_eric_wrapper
+
+BUNDESLAENDER_WITH_STEUERNUMMER = ["BE", "HB", "SH"]
+
+
+def get_bufa_nr(steuernummer: str, bundesland: str, steuernummer_could_be_aktenzeichen=False):
```

Comment:
I'm trying to wrap my head around this `steuernummer_could_be_aktenzeichen`. I'm asking myself at what point can we determine whether the steuernummer being processed fulfills this condition, or in other words, what information is needed to make this determination. In case the information is already available to this function (i.e. `steuernummer` and `bundesland`), we should ideally hide that implementation detail from the caller and simply take care of it here. But I can't answer that question just by looking at the code since right now this flag isn't used anywhere other than in tests, and I couldn't find the answer in the tests either.

I could very well be overlooking something here, but based on what I know right now, I generally recommend implementing features like this not in preparation for an upcoming usage, but together with the usage (in this case that would be an actual caller which sets the flag to true).

Reply:
You are completely correct. I tried to already account use cases (like ESt) that never have aktenzeichen, so I introduced the differentiation, which we don't use.

I actually looked it up again because of your comment and the BUFA-Nr actually is always calculated as id it was a steuernummer, never as an aktenzeichen. So as these are two different functions now for bufa and steuernummer/aktenzeichen generateion, I'll not make a differentiation here.